serving/purgatorio/select-and-train-machine-learning-models/evaluation-and-fine-tunig/evaluation-and-finetuning.md

```diff
@@ -1,5 +1,169 @@
+
+In general, when you build a test set you should make it:
+
+- Big enough to draw conclusions about the performance of the model
+- Representative of the real-world scenario in which the model will be deployed
+- Representative of the training set (if not, for sure you won't get good performances)
+
+Luckily for us, [Andrew NG](https://it.wikipedia.org/wiki/Andrew_Ng) collected for us all a very exhaustive list of tips and trick to build a meaningful test set, and you can find them in the practical book:
+- [Machine Learning Yearning](https://www.deeplearning.ai/machine-learning-yearning/) 
+
+## Select a Metric
+
+You know, Google is your friend, and with a 99.99% chance someone will already be in the same situation as you, and probably a simple query like 'ML metric for **type_X** problem' can give you a ton of good answers. 
+
+A good starting point though can be the following:
+
+- [Selecting Metrics for Machine Learning](https://fayrix.com/machine-learning-metrics)
+
+Even if you can easily find in a lot of literature and good information about each ML metric on Google, be sure to read this presentation about the ML evaluation phase in general: 
+
+- [Performance Evaluation for Learning Algorithms](https://www.icmla-conference.org/icmla11/PE_Tutorial.pdf)
+
+Then you can dive deeper into these more detailed resources:
+
+- [Model Selection and Evaluation](https://frnsys.com/ai_notes/machine_learning/model_selection.html)
+- [Performance Metrics (Error Measures) in Machine Learning Regression](https://arxiv.org/abs/1809.03006)
+
+Bonus resources about model evaluation:
+
+- [Practical Model Evaluation: What matters for your model?](https://www.youtube.com/watch?v=7RdKnACscjA&t=2374s)
+- [Applied ML 2020 - Model Evaluation and Metrics](https://www.youtube.com/watch?v=trg3YkCsjqE)
+
+## Evaluate the Model
+
+After you've built your test set, is time to evaluate your trained algorithm against it!
+
+But to do that, you should have well clear in mind which is a valuable metric to measure the performance of your algorithms. 
+
+It turns out that this heavily depends on the problem you are facing, and in particular the type of problem: are you facing a supervised problem or not? If yes, are you solving a classification or regression problem? 
+
+And so on... 
+
+To learn how to evaluate your model after the training phase, get through Chapter 6 of the book [Machine Learning Engineering](http://www.mlebook.com/wiki/doku.php) by [Andriy Burkov](https://www.linkedin.com/in/andriyburkov/):
+
+- [Chapter 6 - Model Evaluation](https://www.dropbox.com/s/4ly72ahtz6vsy5n/Chapter6.pdf?dl=0)
+
+
+## Hyperparameters Tuning
+
+When you evaluate a Machine Learning model, you evaluate a model trained with a certain configuration, and the parameters in this configuration are called **"hyperparameters"**.
```

Comment:
I can argue here, that tuning can sometimes give a small/marginal gain over the original metric.

But the real factor(s) are two of better metrics/results are:
- quality of the dataset
   - less features the better
   - more focussed and compact features the better
- type of algorithm used

Reply:
cool I add this, thanks